pkg/collector/corechecks/servicediscovery/envs/envs.go

```diff
@@ -35,17 +35,20 @@ type Variables struct {
 
 // NewVariables returns a new [Variables] to collect env. variables.
 func NewVariables(vars map[string]string) Variables {
+	// A regular flow provides an empty input map. For unit tests, verify that the input map has only target variables.
+	for env := range vars {
+		if _, ok := targets[env]; !ok {
+			return Variables{}
+		}
+	}
```

Comment:
a general comment - 

if we have code piece that is relevant only for testing - it should be separated from the "normal" execution flow, in a dedicated function and behind a `test` buildflag

Reply:
I've changed in last commit only unit tests call NewVariables()